Core/include/Acts/Seeding/CandidatesForMiddleSp.hpp

```diff
@@ -0,0 +1,187 @@
+
+  external_space_point_t* bottom{nullptr};
+  external_space_point_t* middle{nullptr};
+  external_space_point_t* top{nullptr};
+  float weight{0.};
+  float zOrigin{0.};
+  bool isQuality{false};
+};
+
+/// @class CandidatesForMiddleSp
+/// The CandidatesForMiddleSp collects the triplet candidates given a
+/// fixed middle spacepoint. It internally stores the triplet candidates
+/// keeping only those with the higher quality.
+///
+/// @tparam external_space_point_t The external spacepoint type.
+
+template <typename external_space_point_t>
+class CandidatesForMiddleSp {
+ public:
+  using value_type = TripletCandidate<external_space_point_t>;
+
+  /// @brief constructor
+  CandidatesForMiddleSp() = default;
+  /// @brief Destructor
+  ~CandidatesForMiddleSp() = default;
+
+  /// @brief Setting maximum number of candidates to keep
+  /// @param n_low Maximum number of candidates in the low-quality collection
+  /// @param n_high Maximum number of candidates in the high-quality collection
+  void setMaxElements(std::size_t n_low, std::size_t n_high);
+
+  /// @brief Retrieve the triplet candidates, the resulting vector is already sorted,
+  /// elements with higher quality first
+  /// @returns Vector of triplet candidates
+  std::vector<value_type> storage();
+
+  /// @brief Adding a new triplet candidate to the collection, should it satisfy the
+  /// selection criteria
+  /// @param SpB Bottom space point
+  /// @param SpM Medium space point
+  /// @param SpT Top space point
+  /// @param weight The quality of the triplet candidate
+  /// @param zOrigin The z-coordinate of the origin
+  /// @param isQuality Whether the triplet candidate is high or low quality
+  void push(external_space_point_t& SpB, external_space_point_t& SpM,
+            external_space_point_t& SpT, float weight, float zOrigin,
+            bool isQuality);
+
+  /// @brief Clear the internal storage
+  void clear();
+
+  /// @brief A function for sorting the triplet candidates from higher to lower quality
+  /// @param i1 First triplet candidate
+  /// @param i2 Second triplet candidate
+  /// @returns The comparison result
+  static bool greaterSort(const value_type& i1, const value_type& i2);
+
+ private:
+  /// @brief dding a new triplet candidate to the collection, should it satisfy the
+  /// selection criteria
+  /// @param storage The collection into which the candidate should be stored
+  /// @param n The current number of stored elements in the container
+  /// @param n_max The maximum number of elements that can be stored in the container
+  /// @param SpB The bottom space point
+  /// @param SpM The middle space point
+  /// @param SpT The top space point
+  /// @param weight The quality of the triplet candidate
+  /// @param zOrigin The z-coordinate of the origin
+  /// @param isQuality Whether the triplet candidate is high or low quality
+  void push(std::vector<std::size_t>& storage, std::size_t& n,
```

Comment:
I wonder if may be worth returning true for added and false otherwise.

Reply:
done